content/Constants/Constants.q.md

```diff
@@ -0,0 +1,1453 @@
+class Circle extends CircularShape
+{
+    public function area()
+    {
+        return self::PI * $this->getRadius() * $this->getRadius();
+    }
+    
+    public function diameter()
+    {
+        return $this->getRadius() * $this->getRadius();
+    }
+    
+    public function circumference()
+    {
+        return 2 * self::PI * $this->getRadius();
+    }
+    
+    public function display()
+    {
+        echo "Radius: " . $this->getRadius();
+        echo "\nDiameter: " . $this->diameter();
+        echo "\nArea: " . $this->area();
+        echo "\nCircumference: " . $this->circumference(); 
+    }
+}
+
+$circleObject = new Circle(3.5);
+$circleObject->display();
+?>
+```
+/// type=SS, answer=[1]
+
+In the statement `return 2 * self::PI * $this->getRadius();` on line 17 of `Circle.php`, replace `self::PI` with `$this->PI`. Execute the program. What is the error message?
+
+ - Undefined property: `Circle::$PI` on line number 17
+
+ - syntax error, unexpected `';'`, expecting `'='` on line number 4
+
+ - Access level to `Circle::area()` must be public (as in class CircularShape) on line number 3
+
+ - Abstract function `CircularShape::diameter()` cannot be declared private on line number 17
+
+ - syntax error, unexpected `'const'` (T_CONST), expecting variable (T_VARIABLE) on line number 4
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[CircularShape.php,Circle.php]
+
+```php
+<?php
+abstract class CircularShape
+{
+    const PI = 3.1416;
+    private $radius;
+    
+    public function __construct($radius)
+    {
+        $this->radius = $radius;
+    }
+    
+    public function getRadius()
+    {
+        return $this->radius;
+    }
+    
+    abstract public function diameter();
+    abstract public function area();
+    abstract public function circumference();
+}
+?>
+```
+
+```php
+<?php
+require_once('./CircularShape.php');
+class Circle extends CircularShape
+{
+    public function area()
+    {
+        return self::PI * $this->getRadius() * $this->getRadius();
+    }
+    
+    public function diameter()
+    {
+        return $this->getRadius() * $this->getRadius();
+    }
+    
+    public function circumference()
+    {
+        return 2 * $this->PI * $this->getRadius();
+    }
+    
+    public function display()
+    {
+        echo "Radius: " . $this->getRadius();
+        echo "\nDiameter: " . $this->diameter();
+        echo "\nArea: " . $this->area();
+        echo "\nCircumference: " . $this->circumference(); 
+    }
+}
+
+$circleObject = new Circle(3.5);
+$circleObject->display();
+?>
+```
+/// type=SS, answer=[3]
+
+In the declaration `public function area()` on line 5 of `Circle.php`, replace `public` with `private`. Execute the program. What is the error message?
+
+ - Undefined property: `Circle::$PI` on line number 17
+
+ - syntax error, unexpected `';'`, expecting `'='` on line number 4
+
+ - Access level to `Circle::area()` must be public (as in class CircularShape) on line number 3
```

Comment:
i tested the code and the result is
' Access level to Circle::area() must be public (as in class CircularShape)'

are back ticks appropriate for CircularShape since it is an abstract class in the program? 'Circle::area()' had back ticks so I thought it might be the same for CircularShape.

Reply:
@clydiex I didn't enclosed CircularShape since this is part of the phrase `as in class CircularShape` which is already enclosed in parentheses. Hope this help. Thanks.

Reply:
noted, thank you.